app/vmalert/notifier/config_watcher.go

```diff
@@ -141,7 +141,7 @@ func targetsFromLabels(labelsFn getLabels, cfg *Config, genFn AlertURLGenerator)
 		}
 		duplicates[u] = struct{}{}
 
-		am, err := NewAlertManager(u, genFn, cfg.HTTPClientConfig, cfg.Timeout.Duration())
+		am, err := NewAlertManager(u, genFn, cfg.HTTPClientConfig, cfg.parsedRelabelConfigs, cfg.Timeout.Duration())
```

Comment:
I think it is incorrect to mix relabel configs for notifier service discovery with relabel configs for labels in fired alerts. It would be better to separate these relabel configs like vmagent and Prometheus does:

* `relabel_configs` contain relabeling rules applied to labels for discovered scrape targets
* `metric_relabel_configs` contain relabeling rules applied to labels for scraped metrics. See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#metric_relabel_configs 

See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#alert_relabel_configs - this looks like what this pull requests needs to implement.

Reply:
Thanks! New config field `alert_relabel_configs` was added in 64cd79a16e9bca960caa2eb5b6cb052f083d2dc8